### PR TITLE
Cldc 955 skip for now button

### PR DESCRIPTION
**Pull request by Ted-U**

### CODE REVIEW

**Comment by baarkerlounger** [comment]
Think the declaration page needs to specifically not have a skip for now link since you have to answer that before you can answer the rest of the subsection.

**Review by baarkerlounger — APPROVED** [review]

File: app/controllers/form_controller.rb
Comment (dushan-madetech) [inline app/controllers/form_controller.rb:132]: What is `calc_questions`?

File: config/forms/2021_2022.json
Comment (dushan-madetech) [inline config/forms/2021_2022.json:1250]: Your editor seems to have changed the formatting/indentation. Or did you deliberately change it?

File: spec/features/form/form_navigation_spec.rb
Comment (dushan-madetech) [inline spec/features/form/form_navigation_spec.rb:75]: You could wrap this in a context block with a description about skip links

File: spec/features/form/form_navigation_spec.rb
Comment (dushan-madetech) [inline spec/features/form/form_navigation_spec.rb:59]: Why did this change?

File: spec/features/form/form_navigation_spec.rb
Comment (dushan-madetech) [inline spec/features/form/form_navigation_spec.rb:114]: This should maybe a context instead of a describe, something like `context "When a user is editing a log" do`

File: spec/features/form/form_navigation_spec.rb
Comment (dushan-madetech) [inline spec/features/form/form_navigation_spec.rb:129]: Given that these lines are the same as those on 135-137 and 146-148 (minus a few minor changes) you could probably extract a method for this

**Review by dushan-madetech — APPROVED** [review]
I left a couple of comments but I don't think it's anything worth blocking this PR over. 🚢 it when you're ready

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]
Number of issues encountered on first pass:

1. ~~I can’t skip the ‘What is the tenancy start date?’ question, instead I am shown validation. If I enter a value to bypass this question and click ‘Skip for now’, then return to the question from the check your answers page, the validation message is shown. Wondering if this might be an issue with all date based questions.~~
2. Changing an answer should show ‘Save changes’ button alongside the ‘Cancel’ link. The button currently says ‘Save and continue’.
3. ~~Clicking ‘Save and continue’ reloads the same page, still showing a ‘Save and continue’ but now alongside a link that says ‘Skip for now’. This is incorrect, it should submit the form with the (changed) value.~~

I can’t test any further as I can’t select a user to create a log. But feels broken, and more testing needed.

File: app/controllers/form_controller.rb
Comment (baarkerlounger) [reply app/controllers/form_controller.rb:132]: It's basically the total charges question (i.e. the one that isn't really asked of users but is the the calculated result of the  sum of the other charges)

File: config/forms/2021_2022.json
Comment (baarkerlounger) [reply config/forms/2021_2022.json:1250]: Was deliberate, made it easier to use code folding to see where to make the changes

File: spec/features/form/form_navigation_spec.rb
Comment (baarkerlounger) [reply spec/features/form/form_navigation_spec.rb:59]: Because "Save and continue" now requires you to provide an answer where previously it didn't and we're not providing one here.

**Comment by baarkerlounger** [comment]
@paulrobertlloyd 

1. Can't reproduce this. Do you have more info?
2. :+1: @Ted-U could you take a look at this?
3. Not sure where you're seeing that? How do you reproduce?

**Comment by paulrobertlloyd** [comment]
Was an issue with an old log with incorrect validation on letting start date. Just issue 2. to resolve.

File: app/models/organisation.rb
Comment (baarkerlounger) [inline app/models/organisation.rb]: Why are we adding `dependent: :destroy` here and below? That seems unrelated to the changes in this PR and I assume has been brought in accidentally from your other branch. I'd suggest removing it from here otherwise gets hard to keep the two changes separate in terms of deploying and reverting.

File: db/schema.rb
Comment (baarkerlounger) [inline db/schema.rb]: Same, would also remove these here as there is no migration in this PR introducing these

File: app/views/form/page.html.erb
Comment (baarkerlounger) [inline app/views/form/page.html.erb:43]: Possibly getting complex enough to be worth extracting to a helper that has a spec?

**Comment by paulrobertlloyd** [comment]
All looks good from my perspective.

**Review by paulrobertlloyd — APPROVED** [review]

**Review by baarkerlounger — CHANGES_REQUESTED** [review]
DB changes needed before this can be merged

File: db/schema.rb
Comment (baarkerlounger) [inline db/schema.rb]: As a general rule of thumb you only want to be committing schema changes if you're also committing a migration that introduces them.

**Review by baarkerlounger — APPROVED** [review]

**Pull request merged**
